src/amp/lib/permutive.ts

```diff
@@ -0,0 +1,57 @@
+export interface PermutivePayload {
+    'properties.content.premium'?: string;
+    'properties.content.id'?: string;
+    'properties.content.title'?: string;
+    'properties.content.section'?: string;
+    'properties.content.authors!'?: string;
+    'properties.content.keywords!'?: string;
+    'properties.content.publishedAt'?: string;
+    'properties.user.edition'?: string;
+}
+
+export const generatePermutivePayload = (
+    rawConfig: CommercialConfigType,
+): PermutivePayload => {
+    const publishedAt =
+        rawConfig.webPublicationDate &&
+        typeof rawConfig.webPublicationDate === 'number'
+            ? new Date(rawConfig.webPublicationDate).toISOString()
+            : null;
+    const authors =
+        rawConfig.author && typeof rawConfig.author === 'string'
+            ? rawConfig.author
+                  .split(',')
+                  .map(s => s.trim())
+                  .join()
+            : null;
+    const keywords =
+        rawConfig.keywords && typeof rawConfig.keywords === 'string'
+            ? rawConfig.keywords
+                  .split(',')
+                  .map(s => s.trim())
+                  .join()
+            : null;
```

Comment:
Could we extract out the repeated code here? Something like:
```typescript
const cleanStringField = (stringField: string): string =>{
    return stringField && typeof stringField === 'string'
            ? stringField
                  .split(',')
                  .map(s => s.trim())
                  .join()
            : null;
}

const authors = cleanStringField(rawConfig.author);
```

Reply:
The `typeof stringField === 'string'` or `typeof rawConfig.keywords === 'string'` aren't needed as we have typescript. Go typescript!

Reply:
@oliverlloyd we don't have a guarantee though as it comes from the frontend config? Correct me if I'm wrong but I was sure TS doesn't check in runtime?!
@buck06191 Great shout, I tend to extract if it's repeated more than 2-3 times and there's no guarantee that `authors` and `keywords` are treated the same in CAPI.